### PR TITLE
Performance Optimizations

### PR DESCRIPTION
**Pull request by RoyalSix**
#### Describe what your pull request addresses (Please include relevant issue numbers):
-

#### Please include detailed Test instructions for your pull request:
-

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6386)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (10af8dd)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-linux-x64-2.0.0-10af8dd.deb) is ready for download.
[macOS build 2.0.0 (10af8dd)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-macos-x64-2.0.0-10af8dd.dmg) is ready for download.
[Windows x64 build 2.0.0 (10af8dd)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x64-2.0.0-10af8dd.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (10af8dd)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x32-2.0.0-10af8dd.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (865ed22)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-linux-x64-2.0.0-865ed22.deb) is ready for download.
[macOS build 2.0.0 (865ed22)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-macos-x64-2.0.0-865ed22.dmg) is ready for download.
[Windows x64 build 2.0.0 (865ed22)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x64-2.0.0-865ed22.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (865ed22)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x32-2.0.0-865ed22.setup.exe) is ready for download.

File: src/js/pages/app.js [hunk 1]
```diff
@@ -19,18 +20,24 @@ import ProjectValidationContainer from '../containers/projectValidation/ProjectV
...
 
 const version = `v${packageJson.version} (${process.env.BUILD})`;
 injectFileLogging(LOG_FILES_PATH, version);
 
 class Main extends Component {
+  constructor(props) {
+    super(props);
+    // load app locale
+    const localeDir = path.join(__dirname, '../../locale');
+    props.loadLocalization(localeDir, props.appLanguage, props.initialize, props.addTranslationForLanguage, props.setActiveLanguage);
```
Comment (mannycolon) [inline src/js/pages/app.js]: These changes should be moved to componentDidMount

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (0ca5575)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-linux-x64-2.0.0-0ca5575.deb) is ready for download.
[macOS build 2.0.0 (0ca5575)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-macos-x64-2.0.0-0ca5575.dmg) is ready for download.
[Windows x64 build 2.0.0 (0ca5575)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x64-2.0.0-0ca5575.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (0ca5575)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x32-2.0.0-0ca5575.setup.exe) is ready for download.

**Comment by RoyalSix** [comment]
Need to update word-map lexer in wA

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (1c659c4)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-linux-x64-2.0.0-1c659c4.deb) is ready for download.
[macOS build 2.0.0 (1c659c4)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-macos-x64-2.0.0-1c659c4.dmg) is ready for download.
[Windows x64 build 2.0.0 (1c659c4)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x64-2.0.0-1c659c4.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (1c659c4)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x32-2.0.0-1c659c4.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (4833346)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-linux-x64-2.0.0-4833346.deb) is ready for download.
[macOS build 2.0.0 (4833346)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-macos-x64-2.0.0-4833346.dmg) is ready for download.
[Windows x64 build 2.0.0 (4833346)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x64-2.0.0-4833346.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (4833346)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x32-2.0.0-4833346.setup.exe) is ready for download.

File: src/js/pages/app.js (hunk 1 above)
Comment (RoyalSix) [reply src/js/pages/app.js]: I was seeing better performance initializing in the constructor

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (412e8cf)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-linux-x64-2.0.0-412e8cf.deb) is ready for download.
[macOS build 2.0.0 (412e8cf)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-macos-x64-2.0.0-412e8cf.dmg) is ready for download.
[Windows x64 build 2.0.0 (412e8cf)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x64-2.0.0-412e8cf.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (412e8cf)](https://tc-builds.door43.org/build/feature-jay-react-localize-3/translationCore-win-x32-2.0.0-412e8cf.setup.exe) is ready for download.

**Pull request merged**